eoflow/models/segmentation_unets.py

```diff
@@ -0,0 +1,446 @@
+#             labels_cropped = tf.image.resize_with_crop_or_pad(labels, out_shape[1], out_shape[2])
+#
+#             if self.config.image_summaries:
+#                 self.add_training_summary(tf.summary.image('input', features[:,0,...][...,0:3]))
+#                 self.add_training_summary(tf.summary.image('labels_raw', labels[...,0:3]))
+#                 self.add_training_summary(tf.summary.image('labels', labels_cropped[...,0:3]))
+#                 self.add_training_summary(tf.summary.image('output', logits[...,0:3]))
+#
+#             # flatten tensors to apply class weighting
+#             flat_logits = tf.reshape(logits, [-1, self.config.n_classes])
+#             flat_labels = tf.reshape(labels_cropped, [-1, self.config.n_classes])
+#
+#             if self.config.class_weights is not None:
+#                 loss = weighted_cross_entropy(flat_logits, flat_labels, self.config.class_weights)
+#             else:
+#                 loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=flat_logits, labels=flat_labels))
+#
+#             self.add_training_summary(tf.summary.scalar('loss', loss))
+#
+#             # update operations for batch-normalisation and define train stepo as minimisation of loss
+#             update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
+#             with tf.control_dependencies(update_ops):
+#                 optimizer = tf.train.AdamOptimizer(self.config.learning_rate)
+#                 train_op = optimizer.minimize(loss,
+#                                               global_step=self.global_step_tensor)
+#
+#             train_head = ModelHeads.TrainHead(train_op, loss, self.get_merged_training_summaries())
+#
+#         if ModelHeads.PREDICT in model_heads:
+#
+#             predictions = {
+#                 'probabilities': probs,
+#                 'predictions': preds
+#             }
+#
+#             predict_head = ModelHeads.PredictHead(predictions)
+#
+#         if ModelHeads.EVALUATE in model_heads:
+#             out_shape = tf.shape(logits)
+#             labels_cropped = tf.image.resize_with_crop_or_pad(labels, out_shape[1], out_shape[2])
+#
+#             labels_n = tf.argmax(labels_cropped[..., 1:], 3)
+#             accuracy_fn = lambda: tf.metrics.accuracy(labels_n, preds)
+#
+#             self.add_validation_metric(accuracy_fn, 'accuracy')
+#
+#             evaluate_ops = self.get_merged_validation_ops()
+#             evaluate_head = ModelHeads.EvaluateHead(*evaluate_ops)
+#
+#         heads = []
+#         for model_head in model_heads:
+#             if model_head == ModelHeads.TRAIN:
+#                 heads.append(train_head)
+#             elif model_head == ModelHeads.PREDICT:
+#                 heads.append(predict_head)
+#             elif model_head == ModelHeads.EVALUATE:
+#                 heads.append(evaluate_head)
+#             else:
+#                 raise NotImplementedError
+#
+#         return heads
```

Comment:
Should we remove this from the code before merging? We can add it back once we have converted the code.